### PR TITLE
*: support "drop global binding"

### PR DESCRIPTION
**Pull request by iamzhoug37**
### What problem does this PR solve? 
associated iusse: https://github.com/pingcap/tidb/issues/8935
feature: support drop global binding
use case:
`
create table t(i int, s varchar(20))
`
`
create index index_t on t(i,s) 
`
`
create global binding for select * from t using select * from t use index for join(index_t)
`

If we want to drop the global binding of `select * from t`, we just execute `drop global binding for select * from t`

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes
N/A

Related changes
N/A

### CODE REVIEW

File: bindinfo/bind_test.go [hunk 1]
```diff
@@ -177,6 +177,20 @@ func (s *testSuite) TestGlobalBinding(c *C) {
...
+	c.Check(err, IsNil)
+	bindData = s.domain.BindHandle().GetBindRecord("select * from t where i > ?", "test")
+	c.Check(bindData, IsNil)
+
+	bindHandle = bindinfo.NewBindHandle(tk.Se, s.Parser)
+	err = bindHandle.Update(true)
+	c.Check(err, IsNil)
+	c.Check(bindHandle.Size(), Equals, 0)
+
+	time.Sleep(time.Second * 6) // wait domain's bindUpdateLoop load deleted bind info.
```
Comment (lamxTyler) [inline bindinfo/bind_test.go]: Better not to sleep so long in the unit test. It will increase the test time and may fail the test.

Comment (iamzhoug37) [reply bindinfo/bind_test.go]: ok

File: bindinfo/handle.go [hunk 2]
```diff
@@ -217,6 +271,36 @@ func (h *BindHandle) appendBindMeta(hash string, meta *bindMeta) {
...
+// removeDeletedBindMeta removes all the bindMeta which originSQL and db are the same with the parameter's meta.
+func (c cache) removeDeletedBindMeta(hash string, meta *bindMeta) {
+	metas, ok := c[hash]
+	if !ok {
+		return
+	}
+
+	for i := len(metas) - 1; i >= 0; i-- {
+		if meta.isSame(meta) {
+			metas = append(metas[:i], metas[i+1:]...)
```
Comment (lamxTyler) [inline bindinfo/handle.go]: If the slice reallocates, the `metas` will be different from `c[hash]`? And so with the `removeStaleBindMetas`.

Comment (lamxTyler) [reply bindinfo/handle.go]: Never mind, remove won't reallocate.

**Pull request merged**